apps/solution/src/components/EditPerson/EditPersonForm.tsx

```diff
@@ -0,0 +1,176 @@
+import { solid } from '@fortawesome/fontawesome-svg-core/import.macro';
+import { Iconized } from '@libs/design';
+import { PersonModel, updatePerson } from '../../api/person';
+import styles from './people.module.scss';
+import { useState } from 'react';
+import useManagers from './UseManagers';
+
+export interface EditPersonFormProps {
+  person: PersonModel;
+  refreshPerson: () => void;
+}
+
+/**
+ *
+ * @param date
+ * @returns new date in format yyyy-mm-dd to satisfie the Input type="date" format
+ */
+const normalizeDate = (date: string): string => {
+  if (date.includes('-')) return date;
+  const [day, month, year] = date.split('/');
+  return `${year}-${month}-${day}`;
+};
+
+/**
+ *
+ * @param date
+ * @returns new date in format dd/mm/yyyy to satisfie the API format
+ */
+const unNormalizeDate = (date: string): string => {
+  if (date.includes('/')) return date;
+  const [year, month, day] = date.split('-');
+  return `${day}/${month}/${year}`;
+};
+
+function EditPersonForm({ person, refreshPerson }: EditPersonFormProps) {
+  const [formData, setFormData] = useState(person);
+  const managers = useManagers(person.id);
+
+  const {
+    firstname,
+    lastname,
+    manager,
+    position,
+    email,
+    phone,
+    photo,
+    entryDate,
+    birthDate,
+    gender,
+    managerId,
+  } = formData;
+
+  const selectedManagerName =
+    managers?.find((m) => m.id === formData.managerId)?.firstname || manager;
+
+  const handleChange = (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    const { name, value } = e.target;
+    console.log(value);
+    setFormData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    updatePerson({
+      ...formData,
+      entryDate: unNormalizeDate(entryDate),
+      birthDate: unNormalizeDate(birthDate),
+      manager: selectedManagerName,
+    }).then(() => refreshPerson());
+  };
```

Comment:
This logic can be extracted in the customHook where refreshPerson() is. Which mean we only call updatePerson(data)